src/solver/reduce/forward_any.cpp

```diff
@@ -0,0 +1,276 @@
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace reduce {
+
+bool AnyForward::IsApplicable(const ExecutionContext& context,
+                              const miopen::reduce::ProblemDescriptionCalculation& problem) const
+{
+    if(!problem.IsValidDim())
+        return false;
+    if(!problem.IsValidLength())
+        return false;
+    if(!problem.IsAllContiguous())
+        return false;
+    if(!IsImprovementOverROCm(context, problem))
+        return false;
+    return true;
+}
+
+ConvSolution
+AnyForward::GetSolution(const ExecutionContext& context,
+                        const miopen::reduce::ProblemDescriptionCalculation& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto dtype       = problem.GetXDesc().GetType();
+    auto input_dtype = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto xdims       = problem.GetXDesc().GetLengths();
+    auto ydims       = problem.GetYDesc().GetLengths();
+    auto dim         = problem.GetDim();
+
+    auto reduce_size = xdims[dim];
+    auto output_numel =
+        std::accumulate(ydims.begin(), ydims.end(), 1ULL, std::multiplies<size_t>());
+
+    auto reqd_work_item_cnt = get_reqd_work_item_cnt(context);
+
+    if(is_parallelism(reqd_work_item_cnt, output_numel, reduce_size))
+    {
+        auto parallelism_size = get_parallelism_size(reqd_work_item_cnt, output_numel, reduce_size);
+
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(parallelism_size * output_numel, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        auto kernel = KernelInfo{};
+
+        kernel.kernel_file = "MIOpenReduceCalculation.cpp";
+        kernel.kernel_name = "CalculationParallelFwdContiguous";
+
+        const auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_INT8", static_cast<int32_t>(dtype == miopenInt8)},
+            {"MIOPEN_USE_FP16", static_cast<int32_t>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int32_t>(dtype == miopenFloat)},
+            {"MIOPEN_USE_BFP16", static_cast<int32_t>(dtype == miopenBFloat16)},
+            {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"OUTPUT_TYPE", input_dtype == "uint8_t" ? "uint8_t" : "bool"},
+            {"OP_TYPE", "ReduceCalculationOp_t::lOR"},
+            {"CALCULATION_DTYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"IS_LOGICAL_CALCULATION", "true"},
+            {"IS_USE_FLOAT_DTYPE", problem.IsValidFloatTypes()},
+            {"MIOPEN_REDUCE_CALCULATION_PROD", MIOPEN_REDUCE_CALCULATION_PROD},
+            {"MIOPEN_REDUCE_CALCULATION_SUM", MIOPEN_REDUCE_CALCULATION_SUM},
+            {"MIOPEN_REDUCE_CALCULATION_ANY", MIOPEN_REDUCE_CALCULATION_ANY},
+            {"MIOPEN_REDUCE_CALCULATION_ALL", MIOPEN_REDUCE_CALCULATION_ALL}};
```

Comment:
if `build_params` at here and below are the same then you should put it outside {} then reuse it

Reply:
solved